Error building with Docker on Linux

**Describe the bug**
When I run `make build-docker-linux` I get a vcs-stamping error. I might have the build environment setup incorrectly? On the chance its not just me... here are the details!

```
make build-docker-linux
docker run \
-v /home/bryan/Projects/lagoon/lagoon-cli/vendor:/go/pkg/mod \
-v /home/bryan/Projects/lagoon/lagoon-cli:/go/src/github.com/uselagoon/lagoon-cli/ \
-e GO111MODULE=on \
-e GOOS=linux \
-e GOARCH=amd64 \
-w="/go/src/github.com/uselagoon/lagoon-cli/" \
golang:1.23 go build -ldflags '-w -s -X github.com/uselagoon/lagoon-cli/cmd.lagoonCLIVersion=v0.31.4+447b638a -X github.com/uselagoon/lagoon-cli/cmd.lagoonCLIBuild=2025-02-07T16:26:43-0600 -X "github.com/uselagoon/lagoon-cli/cmd.lagoonCLIBuildGoVersion=go version go1.23.0 linux/amd64"' -o builds/lagoon-cli-v0.31.4+447b638a-linux-amd64
error obtaining VCS status: exit status 128
        Use -buildvcs=false to disable VCS stamping.
make: *** [Makefile:77: build-docker-linux] Error 1
```

**To Reproduce**
Steps to reproduce the behavior:
1. git clone the lagoon-cli repo, and checkout main branch
2. `make build-docker-linux`
3. See: error obtaining VCS status: exit status 128
        Use -buildvcs=false to disable VCS stamping.

If I add the flag `-buildvcs=false` to the Makefile (see patch attached) I get a successful build using Docker

[Makefile-patched.txt](https://github.com/user-attachments/files/18714352/Makefile-patched.txt)

--- a/.dockerignore
+++ b/.dockerignore
@@ -1,0 +1,6 @@
+Makefile
+mkdocs.yml
+Dockerfile
+Dockerfile.build
+.goreleaser.yml
+builds/

--- a/.gitignore
+++ b/.gitignore
@@ -2,3 +2,4 @@
 /cover.out
 /cover.out.raw
 /sbom.spdx.json
+builds/lagoon

--- a/Dockerfile
+++ b/Dockerfile
@@ -4,12 +4,16 @@ WORKDIR /go/src/github.com/uselagoon/lagoon-cli/
 COPY . .
 
 ARG VERSION
+ARG OS
+ENV OS=${OS:-linux}
+ARG ARCH
+ENV ARCH=${ARCH:-amd64}
 
 RUN apk update && apk add git
 
 RUN VERSION=${VERSION:-"$(echo $(git describe --abbrev=0 --tags)+$(git rev-parse --short=8 HEAD))"} \
 	&& BUILD=$(date +%FT%T%z) \
-  && CGO_ENABLED=0 GOOS=linux go build \
+  	&& CGO_ENABLED=0 GOOS=${OS} GOARCH=${ARCH} go build \
 	-ldflags "-w -s -X github.com/uselagoon/lagoon-cli/cmd.lagoonCLIVersion=$VERSION \
 	-X github.com/uselagoon/lagoon-cli/cmd.lagoonCLIBuild=$BUILD \
 	-X github.com/uselagoon/lagoon-cli/cmd.lagoonCLIBuildGoVersion=go$GOLANG_VERSION" -o lagoon .

--- a/Makefile
+++ b/Makefile
@@ -74,24 +74,12 @@ test-docker:
 
 ## build using docker golang
 build-docker-linux:
-	docker run \
-	-v $(PKGMODPATH):/go/pkg/mod \
-	-v $(DIR):/go/src/${PKG}/ \
-	-e GO111MODULE=on \
-	-e GOOS=linux \
-	-e GOARCH=amd64 \
-	-w="/go/src/${PKG}/" \
-	golang:$(DOCKER_GO_VER) go build -ldflags '${LDFLAGS} -X "${PKG}/cmd.lagoonCLIBuildGoVersion=${GO_VER}"' -o builds/lagoon-cli-${VERSION}-linux-amd64
+	docker build . -t lagoon/lagoon-cli
+	docker run -v $(DIR):/app --entrypoint="/bin/sh" lagoon/lagoon-cli -c "cp /lagoon /app/builds/lagoon"
 
 build-docker-darwin:
-	docker run \
-	-v $(PKGMODPATH):/go/pkg/mod \
-	-v $(DIR):/go/src/${PKG}/ \
-	-e GO111MODULE=on \
-	-e GOOS=darwin \
-	-e GOARCH=amd64 \
-	-w="/go/src/${PKG}/" \
-	golang:$(DOCKER_GO_VER) go build -ldflags '${LDFLAGS} -X "${PKG}/cmd.lagoonCLIBuildGoVersion=${GO_VER}"' -o builds/lagoon-cli-${VERSION}-darwin-amd64
+	docker build . -t lagoon/lagoon-cli-darwin --build-arg OS=darwin --build-arg ARCH=arm64
+	docker run -v $(DIR):/app --entrypoint="/bin/sh" lagoon/lagoon-cli-darwin -c "cp /lagoon /app/builds/lagoon"
 
 install-linux:
 	cp builds/lagoon-cli-${VERSION}-linux-amd64 ${ARTIFACT_DESTINATION}/lagoon

